### ISSUE TITLE
Conflict in Peer Dependencies for @aws-sdk/client-sts  @aws-sdk/client-secrets-manager

### ISSUE THREAD

**Issue by Callenowy** [aws/aws-sdk-js-v3#6098]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

When installing the following packages:
```
npm i @aws-sdk/client-secrets-manager @aws-sdk/client-sts
```
The installation process results in conflicting peer dependencies.
```
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @aws-sdk/credential-provider-web-identity@3.575.0
npm WARN Found: @aws-sdk/client-sts@3.576.0
npm WARN node_modules/@aws-sdk/client-sts
npm WARN   @aws-sdk/client-sts@"*" from the root project
npm WARN   3 more (@aws-sdk/client-secrets-manager, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer @aws-sdk/client-sts@"3.575.0" from @aws-sdk/credential-provider-web-identity@3.575.0
npm WARN node_modules/@aws-sdk/credential-provider-web-identity
npm WARN   @aws-sdk/credential-provider-web-identity@"3.575.0" from @aws-sdk/credential-provider-ini@3.576.0
npm WARN   node_modules/@aws-sdk/credential-provider-ini
npm WARN   1 more (@aws-sdk/credential-provider-node)
npm WARN
npm WARN Conflicting peer dependency: @aws-sdk/client-sts@3.575.0
npm WARN node_modules/@aws-sdk/client-sts
npm WARN   peer @aws-sdk/client-sts@"3.575.0" from @aws-sdk/credential-provider-web-identity@3.575.0
npm WARN   node_modules/@aws-sdk/credential-provider-web-identity
npm WARN     @aws-sdk/credential-provider-web-identity@"3.575.0" from @aws-sdk/credential-provider-ini@3.576.0
npm WARN     node_modules/@aws-sdk/credential-provider-ini
npm WARN     1 more (@aws-sdk/credential-provider-node)
```

### SDK version number

@aws-sdk/client-secrets-manager@3.576.0, @aws-sdk/client-sts@3.576.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

v21.7.2

### Reproduction Steps

1. Run the `npm i @aws-sdk/client-secrets-manager @aws-sdk/client-sts`
2. Observe the warning messages related to peer dependencies.

### Observed Behavior

```
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @aws-sdk/credential-provider-web-identity@3.575.0
npm WARN Found: @aws-sdk/client-sts@3.576.0
npm WARN node_modules/@aws-sdk/client-sts
npm WARN   @aws-sdk/client-sts@"*" from the root project
npm WARN   3 more (@aws-sdk/client-secrets-manager, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer @aws-sdk/client-sts@"3.575.0" from @aws-sdk/credential-provider-web-identity@3.575.0
npm WARN node_modules/@aws-sdk/credential-provider-web-identity
npm WARN   @aws-sdk/credential-provider-web-identity@"3.575.0" from @aws-sdk/credential-provider-ini@3.576.0
npm WARN   node_modules/@aws-sdk/credential-provider-ini
npm WARN   1 more (@aws-sdk/credential-provider-node)
npm WARN
npm WARN Conflicting peer dependency: @aws-sdk/client-sts@3.575.0
npm WARN node_modules/@aws-sdk/client-sts
npm WARN   peer @aws-sdk/client-sts@"3.575.0" from @aws-sdk/credential-provider-web-identity@3.575.0
npm WARN   node_modules/@aws-sdk/credential-provider-web-identity
npm WARN     @aws-sdk/credential-provider-web-identity@"3.575.0" from @aws-sdk/credential-provider-ini@3.576.0
npm WARN     node_modules/@aws-sdk/credential-provider-ini
npm WARN     1 more (@aws-sdk/credential-provider-node)
```

### Expected Behavior

The installation process should resolve peer dependencies without conflicts.

**Comment by kuhe** [comment]
I have confirmed the issue. I have temporarily marked `@aws-sdk/client-sts@3.575.0` as `latest` on NPM to mitigate. 
As of now, the installation command should work.

We intend to release a patch for this by later today.

**Comment by trivikr** [comment]
Reopening to verify fix after https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.577.0 is published, which is expected around 12:30 PM Pacific today.

**Comment by trivikr** [comment]
> verify fix after [`v3.577.0` (release)](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.577.0) is published

The `v3.577.0` is published, and there's no failures with npm install anymore

```console
$ npm i @aws-sdk/client-secrets-manager @aws-sdk/client-sts

$ npm ls @aws-sdk/client-sts
test-npm@1.0.0 /Users/trivikr/workspace/test-npm
├─┬ @aws-sdk/client-secrets-manager@3.577.0
│ ├─┬ @aws-sdk/client-sso-oidc@3.577.0
│ │ └── @aws-sdk/client-sts@3.577.0 deduped
│ ├── @aws-sdk/client-sts@3.577.0 deduped
│ └─┬ @aws-sdk/credential-provider-node@3.577.0
│   ├─┬ @aws-sdk/credential-provider-ini@3.577.0
│   │ └── @aws-sdk/client-sts@3.577.0 deduped
│   └─┬ @aws-sdk/credential-provider-web-identity@3.577.0
│     └── @aws-sdk/client-sts@3.577.0 deduped
└── @aws-sdk/client-sts@3.577.0

$ grep peerDependencies -A 2 node_modules/@aws-sdk/credential-provider-ini/package.json 
  "peerDependencies": {
    "@aws-sdk/client-sts": "^3.577.0"
  },
```

**Comment by Callenowy** [comment]
Beautiful, thank You! :)
